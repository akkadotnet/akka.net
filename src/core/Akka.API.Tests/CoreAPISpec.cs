//-----------------------------------------------------------------------
// <copyright file="CoreAPISpec.cs" company="Akka.NET Project">
//     Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
//     Copyright (C) 2013-2016 Akka.NET project <https://github.com/akkadotnet/akka.net>
// </copyright>
//-----------------------------------------------------------------------

using System;
using System.IO;
using System.Linq;
using System.Runtime.CompilerServices;
using Akka.Cluster;
using Akka.Cluster.Tools.Singleton;
using Akka.Persistence;
using Akka.Remote;
using Akka.Streams.Dsl;
using ApiApprover;
using ApprovalTests;
using Mono.Cecil;
using Xunit;
using Akka.Persistence.Query;

namespace Akka.API.Tests
{
    public class CoreAPISpec
    {
        [Fact]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ApproveCore()
        {
            var assemblyPath = Path.GetFullPath(typeof(PatternMatch).Assembly.Location);
            var asm = AssemblyDefinition.ReadAssembly(assemblyPath);
            var publicApi = Filter(PublicApiGenerator.CreatePublicApiForAssembly(asm));
            Approvals.Verify(publicApi);
        }

        [Fact]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ApproveRemote()
        {
            var assemblyPath = Path.GetFullPath(typeof(RemoteSettings).Assembly.Location);
            var asm = AssemblyDefinition.ReadAssembly(assemblyPath);
            var publicApi = Filter(PublicApiGenerator.CreatePublicApiForAssembly(asm));
            Approvals.Verify(publicApi);
        }

        [Fact]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ApprovePersistence()
        {
            var assemblyPath = Path.GetFullPath(typeof(Persistent).Assembly.Location);
            var asm = AssemblyDefinition.ReadAssembly(assemblyPath);
            var publicApi = Filter(PublicApiGenerator.CreatePublicApiForAssembly(asm));
            Approvals.Verify(publicApi);
        }

        [Fact]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ApprovePersistenceQuery()
        {
            var assemblyPath = Path.GetFullPath(typeof(PersistenceQuery).Assembly.Location);
            var asm = AssemblyDefinition.ReadAssembly(assemblyPath);
            var publicApi = Filter(PublicApiGenerator.CreatePublicApiForAssembly(asm));
            Approvals.Verify(publicApi);
        }

        [Fact]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ApproveCluster()
        {
            var assemblyPath = Path.GetFullPath(typeof(ClusterSettings).Assembly.Location);
            var asm = AssemblyDefinition.ReadAssembly(assemblyPath);
            var publicApi = Filter(PublicApiGenerator.CreatePublicApiForAssembly(asm));
            Approvals.Verify(publicApi);
        }

        [Fact]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ApproveClusterTools()
        {
            var assemblyPath = Path.GetFullPath(typeof(ClusterSingletonManager).Assembly.Location);
            var asm = AssemblyDefinition.ReadAssembly(assemblyPath);
            var publicApi = Filter(PublicApiGenerator.CreatePublicApiForAssembly(asm));
            Approvals.Verify(publicApi);
        }
        [Fact]
        [MethodImpl(MethodImplOptions.NoInlining)]
        public void ApproveStreams()
        {
            var assemblyPath = Path.GetFullPath(typeof(Source).Assembly.Location);
            var asm = AssemblyDefinition.ReadAssembly(assemblyPath);
            var publicApi = Filter(PublicApiGenerator.CreatePublicApiForAssembly(asm));
            Approvals.Verify(publicApi);
        }

        static string Filter(string text)
        {
            return string.Join(Environment.NewLine, text.Split(new[]
            {
                Environment.NewLine
            }, StringSplitOptions.RemoveEmptyEntries)
                .Where(l => !l.StartsWith("[assembly: ReleaseDateAttribute("))
                .Where(l => !string.IsNullOrWhiteSpace(l))
                );
        }

    }
}
