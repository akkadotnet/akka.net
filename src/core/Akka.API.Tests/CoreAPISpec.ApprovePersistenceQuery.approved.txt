[assembly: System.Runtime.InteropServices.ComVisibleAttribute(false)]
[assembly: System.Runtime.InteropServices.GuidAttribute("92ab2788-e008-40d0-8b54-0c95b3cf3404")]
[assembly: System.Runtime.Versioning.TargetFrameworkAttribute(".NETFramework,Version=v4.5", FrameworkDisplayName=".NET Framework 4.5")]
namespace Akka.Persistence.Query
{
    public sealed class EventEnvelope : System.IEquatable<Akka.Persistence.Query.EventEnvelope>
    {
        public EventEnvelope(Akka.Persistence.Query.Offset offset, string persistenceId, long sequenceNr, object @event) { }
        public object Event { get; }
        public Akka.Persistence.Query.Offset Offset { get; }
        public string PersistenceId { get; }
        public long SequenceNr { get; }
        public bool Equals(Akka.Persistence.Query.EventEnvelope other) { }
        public override bool Equals(object obj) { }
        public override int GetHashCode() { }
        public override string ToString() { }
    }
    public interface ICurrentEventsByPersistenceIdQuery : Akka.Persistence.Query.IReadJournal
    {
        Akka.Streams.Dsl.Source<Akka.Persistence.Query.EventEnvelope, Akka.NotUsed> CurrentEventsByPersistenceId(string persistenceId, long fromSequenceNr, long toSequenceNr);
    }
    public interface ICurrentEventsByTagQuery : Akka.Persistence.Query.IReadJournal
    {
        Akka.Streams.Dsl.Source<Akka.Persistence.Query.EventEnvelope, Akka.NotUsed> CurrentEventsByTag(string tag, Akka.Persistence.Query.Offset offset);
    }
    public interface ICurrentPersistenceIdsQuery : Akka.Persistence.Query.IReadJournal
    {
        Akka.Streams.Dsl.Source<string, Akka.NotUsed> CurrentPersistenceIds();
    }
    public interface IEventsByPersistenceIdQuery : Akka.Persistence.Query.IReadJournal
    {
        Akka.Streams.Dsl.Source<Akka.Persistence.Query.EventEnvelope, Akka.NotUsed> EventsByPersistenceId(string persistenceId, long fromSequenceNr, long toSequenceNr);
    }
    public interface IEventsByTagQuery : Akka.Persistence.Query.IReadJournal
    {
        Akka.Streams.Dsl.Source<Akka.Persistence.Query.EventEnvelope, Akka.NotUsed> EventsByTag(string tag, Akka.Persistence.Query.Offset offset);
    }
    public interface IPersistenceIdsQuery : Akka.Persistence.Query.IReadJournal
    {
        Akka.Streams.Dsl.Source<string, Akka.NotUsed> PersistenceIds();
    }
    public interface IReadJournal { }
    public interface IReadJournalProvider
    {
        Akka.Persistence.Query.IReadJournal GetReadJournal();
    }
    public sealed class NoOffset : Akka.Persistence.Query.Offset
    {
        public static Akka.Persistence.Query.NoOffset Instance { get; }
    }
    public abstract class Offset
    {
        protected Offset() { }
        public static Akka.Persistence.Query.Offset NoOffset() { }
        public static Akka.Persistence.Query.Offset Sequence(long value) { }
    }
    public sealed class PersistenceQuery : Akka.Actor.IExtension
    {
        public PersistenceQuery(Akka.Actor.ExtendedActorSystem system) { }
        public Akka.Event.ILoggingAdapter Log { get; }
        public static Akka.Persistence.Query.PersistenceQuery Get(Akka.Actor.ActorSystem system) { }
        public static Akka.Configuration.Config GetDefaultConfig<TJournal>() { }
        public TJournal ReadJournalFor<TJournal>(string readJournalPluginId)
            where TJournal : Akka.Persistence.Query.IReadJournal { }
    }
    public class static PersistenceQueryExtensions
    {
        public static TJournal ReadJournalFor<TJournal>(this Akka.Actor.ActorSystem system, string readJournalPluginId)
            where TJournal : Akka.Persistence.Query.IReadJournal { }
    }
    public class PersistenceQueryProvider : Akka.Actor.ExtensionIdProvider<Akka.Persistence.Query.PersistenceQuery>
    {
        public PersistenceQueryProvider() { }
        public override Akka.Persistence.Query.PersistenceQuery CreateExtension(Akka.Actor.ExtendedActorSystem system) { }
    }
    public sealed class Sequence : Akka.Persistence.Query.Offset, System.IComparable<Akka.Persistence.Query.Sequence>
    {
        public Sequence(long value) { }
        public long Value { get; }
        public int CompareTo(Akka.Persistence.Query.Sequence other) { }
        public override bool Equals(object obj) { }
        public override int GetHashCode() { }
    }
}